src/main/java/TaskManager.java

```diff
@@ -0,0 +1,289 @@
+                    String[] segments = txtLine.split("\\|");
+                    executeCommand(segments[2]);
+                    if (Objects.equals(segments[0], "1")) {
+                        String input = "done " + segments[1];
+                        executeCommand(input);
+                    }
+                    txtLine = br.readLine();
+                }
+            } catch (IOException e) {
+                System.out.println("ERROR OCCURRED");
+            }
+        }
+    }
+
+    public void executeCommand(String inData) {
+        int inDataLength = inData.length();
+        if (!Objects.equals(inData.toLowerCase(), "bye")) {
+            if (Objects.equals(inData, "list")) {
+                listAll();
+            } else if (inData.contains("todo")) {
+                try {
+                    inData.substring(0, 6);
+                    if (Objects.equals(inData.substring(0, 5), "todo ")) {
+                        taskList[i] = new ToDo(inData.substring(5, inDataLength));
+                        toDoAddedMessage();
+                        i++;
+                    }
+                } catch (Exception e) {
+                    errorEmptyMessage("todo");
+                }
+            } else if (inData.contains("deadline ")) {
+                if (Objects.equals(inData.substring(0, 9), "deadline ")) {
+                    String[] segments = inData.split("/by ");
+                    String date = segments[segments.length - 1];
+
+                    int segmentedLength = segments[0].length();
+                    String description = segments[0].substring(9, segmentedLength);
+                    try {
+                        taskList[i] = new Deadline(description, date);
+                        deadlineAddedMessage();
+                        i++;
+                    } catch (Exception e) {
+                        System.out.println("Please enter date in this format: yyyy-mm-dd (e.g., 2019-10-15)");
+                    }
+                }
+            } else if (inData.contains("event ")) {
+                if (Objects.equals(inData.substring(0, 6), "event ")) {
+                    String[] segments = inData.split("/at ");
+                    String date = segments[segments.length - 1];
+
+                    int segmentedLength = segments[0].length();
+                    String description = segments[0].substring(6, segmentedLength);
+                    try {
+                        taskList[i] = new Event(description, date);
+                        eventAddedMessage();
+                        i++;
+                    } catch (Exception e) {
+                        System.out.println("Please enter date in this format: yyyy-mm-dd (e.g., 2019-10-15)");
+                    }
+                }
+            } else if (inData.contains("done ")) {
+                if (isNumeric(inData.substring(5, inDataLength))) {
+                    int taskNo = Integer.parseInt(inData.substring(5, inDataLength));
```

Comment:
Would it be better to call this `taskNumber` to prevent confusion with the negative reply 'No'?